packages/core/src/browser/status-bar/status-bar.tsx

```diff
@@ -109,6 +111,13 @@ export class StatusBarImpl extends ReactWidget implements StatusBar {
         this.node.style.backgroundColor = this.backgroundColor ? this.backgroundColor : null;
     }
 
+    async setColor(color?: string): Promise<void> {
```

Comment:
How about calling `this.update` at the end of the method to trigger rendering? Otherwise, a new color won't be applied immediately.

Reply:
yes sounds good!

Reply:
no, `setBackgroundColor` mutates DOM node directly